### PR TITLE
Add support for implementing encoder acceleration

### PR DESCRIPTION
**Pull request by deltaphi**
This pull requests adds time-tracking between encoder increments. This allows a user of the library to determine the speed of rotation. The speed of rotation can be converted to an acceleration value.

### CODE REVIEW

File: RotaryEncoder.h [hunk 1]
```diff
@@ -1,50 +1,52 @@
...
+#define RotaryEncoder_h
+
+#include "Arduino.h"
+
+#define LATCHSTATE 3
+
+class RotaryEncoder
+{
+public:
+  enum class Direction { NOROTATION = 0, CLOCKWISE = 1, COUNTERCLOCKWISE = 2};
```
Comment (mathertel) [inline RotaryEncoder.h]: Direction::COUNTERCLOCKWISE should be -1 to be compatible with the current version.

Comment (deltaphi) [reply RotaryEncoder.h]: You are right. An additional commit to fix this is available.

**Comment by mathertel** [comment]
I like the additions. Can you provide an example for the timing please ?

**Comment by deltaphi** [comment]
> Can you provide an example for the timing please ?

You mean as in "what does one do with the acceleration values"?

**Comment by mathertel** [comment]
yes, something like: how to implement larger steps when turning faster

**Comment by deltaphi** [comment]
I should be able to lift something from my actual application later today. Unfortunately I don't have the hardware setup to actually runtime-try a limited setup. My hardware involving the encoder is quite integrated.

**Comment by deltaphi** [comment]
Here is an example. I verified that it at least compiles with Arduino 1.8.7 for an Arduino Uno and with PlatformIO <whatever the current version is> for an ESP32 using Arduino.

File: examples/AcceleratedRotator/AcceleratedRotator.ino [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ *   The numbers given in this example are tailored for the following conditions:
+ *   * An encoder with 24 increments per 360 degrees
+ *   * A useful range of 0..1000 rotations
+ *   * Making a 180 degree rotation (12 increments) on the encoder within 50ms will hit
+ *     the opposite end of the range.
+ * 
+ *   Please do not rely on these exact numbers. Recompute them for your usecase and verify
+ *   them with a physical test. A 4ms timing is fairly tight for many controllers. Depending on your
+ *   controller and application, you might not be able to sample the encoder quickly enough to achieve this.
+ */
```
Comment (mathertel) [inline examples/AcceleratedRotator/AcceleratedRotator.ino]: good writing !

**Comment by mathertel** [comment]
Thank you !

**Pull request merged**